### PR TITLE
Zipkin config for setting hostname

### PR DESCRIPTION
**Pull request by esmet**
Allow users to override the hostname used to reach the zipkin collector

### CODE REVIEW

File: source/extensions/tracers/zipkin/zipkin_tracer_impl.cc
Comment (kflynn) [inline source/extensions/tracers/zipkin/zipkin_tracer_impl.cc:81]: Huh. Is it worth reordering the ternary values here to ditch the leading `!`?

File: test/extensions/tracers/zipkin/zipkin_tracer_impl_test.cc
Comment (kflynn) [inline test/extensions/tracers/zipkin/zipkin_tracer_impl_test.cc:101]: Same thing here.

**Review by kflynn — APPROVED** [review]
Looks good! Approving with one question about dropping a negation -- it's OK with me if you don't think it's worth changing. 🙂

**Comment by LukeShu** [comment]
I'm holding off on reviewing this, because last I've heard, we didn't manage to actually replicate the issue, so we haven't been able to confirm that this fixes it?

**Review by LukeShu — APPROVED** [review]

File: source/extensions/tracers/zipkin/zipkin_tracer_impl.cc
Comment (esmet) [reply source/extensions/tracers/zipkin/zipkin_tracer_impl.cc:81]: Keeping it as it is, since either order is valid in its own way

**Pull request merged**
